pkg/kube/controllers/quarkssecret/quarkssecret_reconciler.go

```diff
@@ -239,6 +250,70 @@ func (r *ReconcileQuarksSecret) createRSASecret(ctx context.Context, qsec *qsv1a
 	return r.createSecrets(ctx, qsec, secret)
 }
 
+// renderSecret uses the specified templating engine to draw the secret data
+func (r *ReconcileQuarksSecret) renderSecret(ctx context.Context, qsec *qsv1a1.QuarksSecret) (map[string]string, error) {
+	res := map[string]string{}
+	if qsec.Spec.TemplateType == "" {
+		return res, errors.New("templatedConfig needs a templateType to be specified. E.g. helm")
+	}
+
+	// Interpolate the given values, and retrieve the secret contents to fill our config with
+	for k, v := range qsec.Spec.TemplateValues {
+		fields := strings.Split(v.(string), ".")
+		if len(fields) != 2 {
+			return res, errors.New("failed while reading templatedConfig values. Values must be in the `secretname.field` format. Got: " + v.(string))
+		}
+		secretName := fields[0]
+		field := fields[1]
+
+		existingSecret := &corev1.Secret{}
+		namespacedName := types.NamespacedName{
+			Namespace: qsec.Namespace,
+			Name:      secretName,
+		}
+		err := r.client.Get(ctx, namespacedName, existingSecret)
+		if err != nil {
+			if apierrors.IsNotFound(err) {
+				return res, newSecNotReadyError("secret not found")
+			}
+			return res, errors.Wrap(err, "getting secret")
+		}
+		data, ok := existingSecret.Data[field]
+		if !ok {
+			return res, errors.Errorf("Failed to get secret data key: %s", field)
+		}
+		qsec.Spec.TemplateValues[k] = string(data)
```

Comment:
I'd rather use a new map instead of overwriting the references in the qsec struct